<launch>
 
    <!-- Choose visualization -->
 
  <arg name="compressed"                  default="true"/>
  <arg name="rgbd_topic"                  default="/rtabmap/rgbd_image"/>
  <arg  name="rgbd_topic_relay"          default="$(arg rgbd_topic)_relay"/>



   <node pkg="wavnodes" type="POSE2ODOM" name="POSE2ODOM">
    <param name="pub_odom" type="bool" value="true" />
    <param name="pub_odom_tf" type="bool" value="false" />
   </node>




 <group ns="rtabmap">

    <!-- Odometry -->
     <node name="republish_rgbd_image"  type="rgbd_relay" pkg="rtabmap_ros">
	   <remap     from="rgbd_image" to="$(arg rgbd_topic)/compressed"/>
	   <remap     from="$(arg rgbd_topic)/compressed_relay" to="$(arg rgbd_topic_relay)"/>
	   <param  name="uncompress" value="true"/>
     </node>


</group>

<node pkg="wavnodes" type="RGBD2PCL" name="RGBD2PCL" />

<include file="$(find wavnodes)/launch/RGBD2PCL.launch"/>

<node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
	   
	    <param name="map_frame" value="odom" />
	    <param name="base_frame" value="base_link" />
	    <param name="odom_frame" value="odom" />
	    
	    <param name="pub_map_odom_transform" value="false"/>
	    <param name="pub_map_scanmatch_transform" value="false"/>
	    <param name="pub_odometry" value="true"/>
	    
	    <param name="map_resolution" value="0.05"/>
	    <param name="map_size" value="2048"/>
	    <param name="map_multi_res_levels" value="2" />
	    
	    <param name="laser_min_dist" value="0.20" />
	    <param name="laser_max_dist" value="8.0" />
	    <param name="map_update_angle_thresh" value="0.018" />
	    <param name="map_update_distance_thresh" value="0.001" />
	    
	    <param name="scan_topic" value="/rtabmap/scan"/>
	  </node>



  <node pkg="robot_localization" 
        type="ukf_localization_node" 
        name="ukf_localization"
        clear_params="true"
        output="screen">
        <param name="frequency" value="5.0" />
        <param name="sensor_timeout" value="10" />
        <param name="map_frame" value="map" />
        <param name="odom_frame" value="odom" />
        <param name="base_link_frame" value="base_link" />
        <param name="transform_time_offset" value="0.0" />
        <param name="two_d_mode" value="true" />

        

        <param name="odom0" value="/odom" />
        <param name="odom1" value="/rtabmap/odom" />
        <param name="odom2" value="/wav_laser_odom" />



        <rosparam param="odom0_config">
         [true, true, false,
          false, false, true,
          false, false, false,
          false, false, false,
          false, false, false]</rosparam>
    
        <!-- <rosparam param="odom1_config">
         [true, true, false,
          false, false, true,
          false, false, false,
          false, false, false,
          false, false, false]</rosparam>

        <rosparam param="odom2_config">
         [true, true, false,
          false, false, true,
          false, false, false,
          false, false, false,
          false, false, false]</rosparam> -->

        <param name="odom0_differential" value="false" />
        <param name="odom0_relative" value="true" />
        <param name="odom1_differential" value="true" />
        <param name="odom1_relative" value="false" />
        <param name="odom2_differential" value="true" />
        <param name="odom2_relative" value="false" />

        <param name="print_diagnostics" value="true" />

        <param name="odom0_queue_size" value="10" /> 
        <param name="odom1_queue_size" value="10" />
        <param name="odom2_queue_size" value="5" /> 

  </node>  


 <!-- <node pkg="robot_localization" 
        type="ukf_localization_node" 
        name="ukf_localization"
        clear_params="true"
        output="screen">
        <param name="frequency" value="5.0" />
        <param name="sensor_timeout" value="10" />
        <param name="map_frame" value="map" />
        <param name="odom_frame" value="odom" />
        <param name="base_link_frame" value="base_link" />
        <param name="transform_time_offset" value="0.0" />
        <param name="two_d_mode" value="true" />

        

        <param name="odom0" value="/odom" />
        <param name="odom1" value="/rtabmap/odom" />



        <rosparam param="odom0_config">
         [true, true, false,
          false, false, true,
          false, false, false,
          false, false, false,
          false, false, false]</rosparam>
    



        <param name="odom0_differential" value="false" />
        <param name="odom0_relative" value="true" />
        <param name="odom1_differential" value="true" />
        <param name="odom1_relative" value="false" />
        <param name="odom2_differential" value="true" />
        <param name="odom2_relative" value="false" />

        <param name="print_diagnostics" value="false" />

        <param name="odom0_queue_size" value="10" /> 
        <param name="odom1_queue_size" value="10" />
        <param name="odom2_queue_size" value="5" /> 

  </node> -->

 
</launch>
